internal/resource/config/session/applicationsessionpolicy/session_application_session_policy_data_source.go

```diff
@@ -0,0 +1,106 @@
+type sessionApplicationSessionPolicyDataSource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type sessionApplicationSessionPolicyDataSourceModel struct {
+	Id              types.String `tfsdk:"id"`
+	IdleTimeoutMins types.Int64  `tfsdk:"idle_timeout_mins"`
+	MaxTimeoutMins  types.Int64  `tfsdk:"max_timeout_mins"`
+}
+
+// GetSchema defines the schema for the datasource.
+func (r *sessionApplicationSessionPolicyDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Describes a SessionApplicationSessionPolicy.",
+		Attributes: map[string]schema.Attribute{
+			// Add necessary attributes here
+			"idle_timeout_mins": schema.Int64Attribute{
+				Description: "The idle timeout period, in minutes. If set to -1, the idle timeout will be set to the maximum timeout. The default is 60.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"max_timeout_mins": schema.Int64Attribute{
+				Description: "The maximum timeout period, in minutes. If set to -1, sessions do not expire. The default is 480.",
+				Computed:    true,
+				Optional:    true,
+			},
+		},
+	}
+
+	id.ToSchema(&schema)
+	resp.Schema = schema
+}
+
+// Metadata returns the datasource type name.
+func (r *sessionApplicationSessionPolicyDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_session_application_session_policy"
+}
+
+func (r *sessionApplicationSessionPolicyDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, _ *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readSessionApplicationSessionPolicyResponseDataSource(ctx context.Context, r *client.ApplicationSessionPolicy, state *sessionApplicationSessionPolicyDataSourceModel) {
+	state.Id = types.StringValue("sessionApplicationSessionPolicyId")
+	state.IdleTimeoutMins = types.Int64Value(r.GetIdleTimeoutMins())
+	state.MaxTimeoutMins = types.Int64Value(r.GetMaxTimeoutMins())
+}
+
+func (r *sessionApplicationSessionPolicyDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state sessionApplicationSessionPolicyDataSourceModel
+
+	diags := req.Config.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiReadSessionApplicationSessionPolicy, httpResp, err := r.apiClient.SessionAPI.GetApplicationPolicy(config.ProviderBasicAuthContext(ctx, r.providerConfig)).Execute()
+	// Read the response into the state
+	if err != nil {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while getting the Local Identity Profile", err, httpResp)
```

Comment:
Wrong data source name in this error message

Reply:
corrected